### PR TITLE
Add deployment debug setting

### PR DESCRIPTION
**Pull request by vivsriaus**

### CODE REVIEW

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentDebugDetailLevel.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.Linq;
+
+namespace Microsoft.Azure.Management.Resources.Models
+{
+    /// <summary>
+    /// Specifies the deployment debug detail level for the deployment
+    /// operations.
+    /// </summary>
+    [Flags]
+    public enum DeploymentDebugDetailLevel
```
Comment (ravbhatnagar) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentDebugDetailLevel.cs]: IMO, we call it DeploymentDebugLogLevel to keep it consistent across clients. Also, "All" is a valid option when someone wants to specify both request and response log levels.

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentDebugDetailLevel.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        None = 0,
+        
+        RequestContent = 1,
```
Comment (ravbhatnagar) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentDebugDetailLevel.cs]: HttpRequestContent and HttpReponseContent?

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentDebugDetailLevel.cs (hunk 1 above)
Comment (vivsriaus) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentDebugDetailLevel.cs]: Spoke to @ravbhatnagar This is what we have in source, and it looks like the pattern for naming other objects is similar - adhere closely to the source code. So, we'll keep this as is unless there is a strong indication that this name won't be clear for users

**Comment by vivsriaus** [comment]
@hovsepm @markcowl @stankovski Can one of you please review and merge this PR? Thanks.

**Pull request merged**